src/target/riscv/riscv.c

```diff
@@ -2697,7 +2768,158 @@ COMMAND_HANDLER(handle_repeat_read)
 	return result;
 }
 
+COMMAND_HANDLER(handle_memory_sample_command)
+{
+	struct target *target = get_current_target(CMD_CTX);
+
+	if (CMD_ARGC == 0) {
+		for (unsigned i = 0; i < DIM(sample_config.bucket); i++) {
+			if (sample_config.bucket[i].enabled) {
+				command_print(CMD, "bucket %d; address=0x%" TARGET_PRIxADDR "; size=%d", i,
+							  sample_config.bucket[i].address,
+							  sample_config.bucket[i].size_bytes);
+			} else {
+				command_print(CMD, "bucket %d; disabled", i);
+			}
+		}
+		return ERROR_OK;
+	}
+
+	if (CMD_ARGC < 2) {
+		LOG_ERROR("Command requires at least bucket and address arguments.");
+		return ERROR_FAIL;
```

Comment:
If you'd like, you can return `ERROR_COMMAND_SYNTAX_ERROR`, which makes OpenOCD print the syntax help for the command.

Reply:
Done.